### PR TITLE
Update datasets-download-stats.md

### PR DESCRIPTION
**Pull request by lhoestq**
Updated the download count method, and I kept how it was working before september 2024 (since odler data an be viewer from Enterprise analytics) cc @julien-c

### CODE REVIEW

File: docs/hub/datasets-download-stats.md [hunk 1]
```diff
@@ -2,7 +2,11 @@
 
 ## How are download stats generated for datasets?
 
-The Hub provides download stats for all datasets loadable via the `datasets` library. To determine the number of downloads, the Hub counts every time `load_dataset` is called in Python, excluding Hugging Face's CI tooling on GitHub. No information is sent from the user, and no additional calls are made for this. The count is done server-side as we serve files for downloads. This means that:
+Counting the number of downloads for datasets is not a trivial task, as a single dataset repository might contain multiple files, from multiple subsets and splits (e.g. train/validation/test) and sometimes with many files in a single split. To avoid double counting downloads (e.g., counting a single download of a dataset as multiple downloads), the Hub counts as one download every series of files requests in an interval of 5 minutes per user. No information is sent from the user, and no additional calls are made for this. The count is done server-side as the Hub serves files for downloads. Every HTTP request to the files, including `GET` and `HEAD`, will be counted as a download.
```
Comment (davanstrien) [inline docs/hub/datasets-download-stats.md:5]:
```suggestion
Counting the number of downloads for datasets is not a trivial task, as a single dataset repository might contain multiple files, from multiple subsets and splits (e.g. train/validation/test) and sometimes with many files in a single split. To solve this issue and avoid counting one person's download multiple times, we treat all files downloaded by a user within a 5-minute window as a single dataset download. This counting happens automatically on our servers when files are downloaded (through GET or HEAD requests), with no need to collect any user information or make additional calls.
```

Comment (davanstrien) [reply docs/hub/datasets-download-stats.md:5]: I tried to make this a bit more clear but feel free to use the previous version

File: docs/hub/datasets-download-stats.md [hunk 2]
```diff
@@ -2,7 +2,11 @@
...
 
-* The download count is the same regardless of whether the data is directly stored on the Hub repo or if the repository has a [script](/docs/datasets/dataset_script) to load the data from an external source.
-* If a user manually downloads the data using tools like `wget` or the Hub's user interface (UI), those downloads will not be included in the download count.
+## Before Setpember 2024
+
+The Hub used to provide download stats only for the datasets loadable via the `datasets` library. To determine the number of downloads, the Hub used to count every time `load_dataset` is called in Python, excluding Hugging Face's CI tooling on GitHub. No information was sent from the user, and no additional calls were made for this. The count was done server-side as we served files for downloads. This means that:
```
Comment (davanstrien) [inline docs/hub/datasets-download-stats.md:9]:
```suggestion
The Hub used to provide download stats only for the datasets loadable via the `datasets` library. To determine the number of downloads, the Hub previously counted every time `load_dataset` was called in Python, excluding Hugging Face's CI tooling on GitHub. No information was sent from the user, and no additional calls were made for this. The count was done server-side as we served files for downloads. This means that:
```

**Review by davanstrien — APPROVED** [review]

File: docs/hub/datasets-download-stats.md (hunk 1 above)
Comment (lhoestq) [reply docs/hub/datasets-download-stats.md:5]: ah nice ! your suggestion is clearer, thanks

File: docs/hub/datasets-download-stats.md [hunk 3]
```diff
@@ -2,7 +2,11 @@
...
+Counting the number of downloads for datasets is not a trivial task, as a single dataset repository might contain multiple files, from multiple subsets and splits (e.g. train/validation/test) and sometimes with many files in a single split. To solve this issue and avoid counting one person's download multiple times, we treat all files downloaded by a user within a 5-minute window as a single dataset download. This counting happens automatically on our servers when files are downloaded (through GET or HEAD requests), with no need to collect any user information or make additional calls.
 
-* The download count is the same regardless of whether the data is directly stored on the Hub repo or if the repository has a [script](/docs/datasets/dataset_script) to load the data from an external source.
-* If a user manually downloads the data using tools like `wget` or the Hub's user interface (UI), those downloads will not be included in the download count.
+## Before Setpember 2024
```
Comment (julien-c) [inline docs/hub/datasets-download-stats.md:7]: *september

File: docs/hub/datasets-download-stats.md [hunk 4]
```diff
@@ -2,7 +2,11 @@
 
 ## How are download stats generated for datasets?
```
Comment (julien-c) [inline docs/hub/datasets-download-stats.md:3]: would be clearer maybe to word it like: "How are downloads counted for datasets" (same for models)

File: docs/hub/datasets-download-stats.md [hunk 5]
```diff
@@ -2,7 +2,11 @@
...
 
-The Hub provides download stats for all datasets loadable via the `datasets` library. To determine the number of downloads, the Hub counts every time `load_dataset` is called in Python, excluding Hugging Face's CI tooling on GitHub. No information is sent from the user, and no additional calls are made for this. The count is done server-side as we serve files for downloads. This means that:
+Counting the number of downloads for datasets is not a trivial task, as a single dataset repository might contain multiple files, from multiple subsets and splits (e.g. train/validation/test) and sometimes with many files in a single split. To solve this issue and avoid counting one person's download multiple times, we treat all files downloaded by a user within a 5-minute window as a single dataset download. This counting happens automatically on our servers when files are downloaded (through GET or HEAD requests), with no need to collect any user information or make additional calls.
```
Comment (julien-c) [inline docs/hub/datasets-download-stats.md:5]: > by a user (based on their IP address)

maybe add this to make it clearer for us? @lhoestq

**Pull request merged**